systemlink/clients/tag/_tag_selection.py

```diff
@@ -338,11 +338,18 @@ def close(self) -> None:
 
         self._close_internal()
         self._closed = True
-        self._paths.clear()
-        self._metadata.clear()
-        self._readers.clear()
-        if self._values is not None:
-            self._values.clear()
+        paths = getattr(self, "_paths", None)
+        if paths is not None:
+            paths.clear()
+        metadata = getattr(self, "_metadata", None)
+        if metadata is not None:
+            metadata.clear()
+        readers = getattr(self, "_readers", None)
+        if readers is not None:
+            readers.clear()
+        values = getattr(self, "_values", None)
+        if values is not None:
+            values.clear()
```

Comment:
Some of the tests had warnings because this `close()` method was throwing an exception (called from `__del__` as a part of GC). The offending tests were verifying that the constructor was erroring on bad arguments, but this meant that some attributes weren't ever being set. I made `close()` safer by checking for attributes before using them.